src/main/java/duke/ui/ListLoader.java

```diff
@@ -0,0 +1,40 @@
+package duke.ui;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.ArrayList;
+
+import duke.tasks.Task;
+import duke.tasks.TaskManager;
+
+/**
+ * ListLoader Object that handles loading of task list from
+ * a saved ser file during program initialisation.
+ */
+public class ListLoader {
+    private TaskManager taskManager;
+
+    /**
+     * Constructs ListLoader Object.
+     *
+     * @param manager TaskManager to pass saved task list elements
+     */
+    public ListLoader(TaskManager manager) {
+        this.taskManager = manager;
+    }
+
+    /**
+     * Loads saved task list into TaskManager Object.
+     *
+     * @throws IOException if ser file does not exist
+     * @throws ClassNotFoundException if Task Objects in stored list are not serializable
+     */
+    public void loadList() throws IOException, ClassNotFoundException {
+        FileInputStream readData = new FileInputStream("storage/list.ser");
+        ObjectInputStream readStream = new ObjectInputStream(readData);
+        ArrayList<Task> taskList = (ArrayList<Task>) readStream.readObject();
```

Comment:
Perhaps consider using a plural noun as the name of the variable?

Reply:
Hmm I think I'll stick to tasklist since it keeps the purpose very clear on what it's supposed to hold, but thanks!